### PR TITLE
Jordgubb gfx

### PR DESCRIPTION
**Pull request by FredTheDino**
Add in the gfx layer

### CODE REVIEW

**Review by EMattfolk — COMMENTED** [review]
I don't like the changes to `unify` but the rest looks ok.
I wondered if we could turn some of the longer function calls into builders. It may give a nice structure. Something like:
```
image.draw_builder()
     .position(0, 0)
     .rotation(3.14)
     .tint(Color{...})
     .draw()
```

File: sylt-compiler/src/typechecker.rs [hunk 1]
```diff
@@ -1463,13 +1463,31 @@ impl TypeChecker {
...
 
-        if a == b {
+        if a == b || seen.contains(&(a, b)) {
             return Ok(a);
         }
 
+        // Equivalence is symetrical!
+        seen.insert((a, b));
+        seen.insert((b, a));
+
```
[the remark this answers is not in the document]
Comment (FredTheDino) [reply sylt-compiler/src/typechecker.rs:1490]: ?
Why would it not be doing anything? Why is this a bug?

File: sylt-compiler/src/typechecker.rs [hunk 2]
```diff
@@ -1463,13 +1463,31 @@ impl TypeChecker {
...
+    }
+
+    fn sub_unify(
+        &mut self,
+        span: Span,
+        ctx: TypeCtx,
+        a: usize,
+        b: usize,
+        seen: &mut BTreeSet<(usize, usize)>,
+    ) -> TypeResult<usize> {
```
[the remark this answers is not in the document]
Comment (FredTheDino) [reply sylt-compiler/src/typechecker.rs:1479]: How do you know?

**Review by EMattfolk — APPROVED** [review]
I still doesn't sit well with me that we have to add special cases to the unification, though I will accept it since it seems like a reasonable fix given the time constraints.

Also it is spelled *infinite* :wink:

**Review by sornas — APPROVED** [review]
some comments are over at #585 since they're entangled

**Pull request merged**